halibot/halauth.py

```diff
@@ -0,0 +1,77 @@
+import json
+import logging
+
+def hasPermission(perm, reply=False):
+	def real_dec(func):
+		def wrapper(self, msg, *args, **kwargs):
+			if self._hal.auth.hasPermission(msg.origin, msg.author, perm):
+				func(self, msg, *args, **kwargs)
+			elif reply:
+				self.reply(msg, body="Permission Denied")
```

Comment:
Something like that should be fine. Something like `perm-denied-msg` might be better, but I am not too picky. When it comes to config v. annotation argument for this and for the `reply` bool, could have them in the config and also have an optional argument that when set overrides the config setting.